library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
I would've closed this PR somewhere in the middle of this quite pointless
conversation.
On Вт, 24 марта 2015 г. at 9:28 Evgeniy Polyakov <notifications@github.com>
wrote:

> In library/index.c
> <https://github.com/reverbrain/eblob/pull/107#discussion_r27026477>:
>
> >
> >  	if (b == NULL || bctl == NULL)
> >  		return -EINVAL;
> >
> > +	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
> > +	cache_is_empty = eblob_cache_empty(b);
> > +
>
> Because I suspect that turning off binary log during sorting is dangerous.
> It looks like it is safe, but there are only 2 major changes which affect
> data consistency and it is one of them.
> Since there is data corruption bug I had rather removed suspicious commit
> than doing nothing.
>
> To make things clear - I will implement check which will skip cache
> flushing, this should make initial blob loading as fast as it was.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/reverbrain/eblob/pull/107/files#r27026477>.
>